Modules/Core/Common/test/itkDirectoryTest.cxx

```diff
@@ -45,14 +45,8 @@ itkDirectoryTest(int argc, char * argv[])
     std::cout << "File 0 is " << directory->GetFile(0) << std::endl;
   }
 
-  // This should fail
-  auto fileOutOfRange = static_cast<unsigned int>(directory->GetNumberOfFiles());
-  if (directory->GetFile(fileOutOfRange))
-  {
-    std::cerr << "directory->GetFile(directory->GetNumberOfFiles())"
-              << " should have failed." << std::endl;
-    return EXIT_FAILURE;
-  }
+  // Since 78dcc270826fe87fac5bf46d65ca6f02d9f47b4b the next line causes a crash:
+  // auto fileOutOfRange = static_cast<unsigned int>(directory->GetNumberOfFiles());
```

Comment:
There is no reason to support out-of-bounds indexing, so I suggest just revising this commit to remove that test case.